.github/workflows/e2e.yml

```diff
@@ -0,0 +1,101 @@
+name: E2E
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+env:
+  FLWR_TELEMETRY_ENABLED: 0
+
+jobs:
+  pytorch:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.7.15
+      - name: Install
+        run: |
+          python -m pip install -U pip==22.3.1
+          python -m pip install -U setuptools==65.6.3
+          python -m pip install poetry==1.3.2
+          poetry config virtualenvs.create false
+      - name: Install dependencies
+        run: |
+          cd e2e/pytorch
+          python -m poetry install
+      - name: Cache Datasets
+        uses: actions/cache@v2
+        with:
+          path: "./e2e/pytorch/data"
+          key: pytorch-datasets
+      - name: Download Datasets
+        run: |
+          cd e2e/pytorch
+          python -c "from torchvision.datasets import CIFAR10; CIFAR10('./data', download=True)"
```

Comment:
@charlesbvll, how about running the cache@v3 and checking for a cache hit before triggering the download once more?

=> https://github.com/actions/cache/tree/main#restoring-and-saving-cache-using-a-single-action

It might reduce flaky jobs in case the remote servers are not reachable.

Reply:
@charlesbvll I am merging it. Feel free to do this in a follow-up PR if you think its worth it.